### PR TITLE
Fix duplicate migrations bug

### LINKED ISSUE #1935: `makemigrations` always creates a new migration
Multiple people reported that running `docker exec mathesar_service sh -c "python manage.py makemigrations"` always creates a new migration, even if you run it multiple times in sequence, that looks like this:

```
# Generated by Django 3.1.14 on 2022-11-15 13:18

from django.db import migrations, models
import mathesar.models.query


class Migration(migrations.Migration):

    dependencies = [
        ('mathesar', '0007_auto_20221115_1318'),
    ]

    operations = [
        migrations.AlterField(
            model_name='uiquery',
            name='display_options',
            field=models.JSONField(blank=True, null=True, validators=[mathesar.models.query._get_validator_for_dict]),
        ),
        migrations.AlterField(
            model_name='uiquery',
            name='initial_columns',
            field=models.JSONField(validators=[mathesar.models.query._get_validator_for_list_of_dicts, mathesar.models.query._get_validator_for_initial_columns]),
        ),
        migrations.AlterField(
            model_name='uiquery',
            name='transformations',
            field=models.JSONField(blank=True, null=True, validators=[mathesar.models.query._get_validator_for_list_of_dicts, mathesar.models.query._get_validator_for_transformations]),
        ),
    ]
```

Subsequent runs create migrations that have the previous migration as its dependency. That's the only difference between these redundant migrations.

The STDOUT is:

```
Migrations for 'mathesar':
  mathesar/migrations/0008_auto_20221115_1318.py
    - Alter field display_options on uiquery
    - Alter field initial_columns on uiquery
    - Alter field transformations on uiquery
```

### PR DESCRIPTION
**Pull request by aagman945**
Fixes #1935 

**Technical details**

- replaced custom validators with class-based validators 

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by kgodey** [comment]
Assigned this to @dmos62 since he created the issue and @silentninja for Django familiarity.

File: mathesar/models/query.py [hunk 1]
```diff
@@ -91,33 +112,53 @@ def _validator(jp_path):
...
                         raise InvalidValueType(
                             message,
-                            field=field_name,
+                            field=self.field_name,
                         )
-    return _validator
 
+    def __eq__(self, other):
+        return (
+            isinstance(other, JpPathValidator) and self.field_name == other.field_name and self.jp_path == other.jp_path
```
Comment (dmos62) [inline mathesar/models/query.py:120]: Is `and self.jp_path == other.jp_path` a mistake? Doesn't seem to be defined, and I'm not sure what the intent could be.

**Review by dmos62 — CHANGES_REQUESTED** [review]
I didn't know about `django.utils.deconstruct.deconstructible`. Kudos!

File: mathesar/models/query.py (hunk 1 above)
Comment (aagman945) [reply mathesar/models/query.py:120]: mybad its a mistake

**Comment by aagman945** [comment]
@dmos62 we should move validators to `validaros.py`. Also we should generate new migrations since there are a lot of duplicate migrations

**Comment by dmos62** [comment]
@aagman945 True. Feel free to move them to a separate file. Can we generate migrations in a way that wouldn't require the rest of us to wipe our databases or to take additional steps following the merging of this PR? If not, we're planning to reset before launch.

**Comment by aagman945** [comment]
@dmos62 its better to reset migrations

**Comment by kgodey** [comment]
We have an issue to reset migrations before release: #202, so it's fine if we need to reset them.

**Review by silentninja — APPROVED** [review]
Nice work! @aagman945

**Review by dmos62 — APPROVED** [review]

**Comment by dmos62** [comment]
Tested. Enabling auto-merge. Let's reset migrations in a dedicated PR.

**Pull request merged**